server/server.js

```diff
@@ -41,6 +43,23 @@ if (process.env.ENV === "PRODUCTION") {
   });
 }
 
+app.get("/authenticate/user-posts", (req, res) => {
```

Comment:
adding on, can we make this code use async await? I see below you're using it for a user promise, but maybe we can use it for user as well?
```suggestion
app.get("/authenticate/user-posts", async (req, res) => {
```